### DISCUSSION TITLE
Spaceship's ConnectAPI::Token.create() token be used for Tunes calls?

### DISCUSSION THREAD

**Discussion by tanya355** [fastlane/fastlane#17508] [Q&A] [answered]
Hi 👋 ,

I'm trying to find a way to authenticate the Spaceship IAP creation, updation and fetch methods. 
i.e. I'll be using the following methods:
```
app = Spaceship::Application.find()
app.in_app_purchases.create()
app.in_app_purchases.all.first
```
I wanted know if there was a way I could use the token generated like below for those `Spaceship::Tunes` functions:
```
token = Spaceship::ConnectAPI::Token.create(
  key_id: 'the-key-id',
  issuer_id: 'the-issuer-id',
  filepath:  File.absolute_path("../AuthKey_the-key-id.p8")
)

Spaceship::ConnectAPI.token = token
```
I did notice that the token file was within connect_api folder `spaceship/lib/spaceship/connect_api/token.rb`
Is the only way to use those first mentioned functions with a username & password login?

Thank you for reading!

**Comment by rogerluan** [comment, marked as answer]
Hey @tanya355 ! 👋 

>Is the only way to use those first mentioned functions with a username & password login?

Unfortunately, yes. Apple new set of APIs doesn't support IAP yet (AFAICT).

Hope this helps!
